timemachine/fe/dummy.py

```diff
@@ -91,19 +102,18 @@ def identify_root_anchors(mol, core, dummy_atom):
         List of root anchors that the dummy atom is connected to.
 
     """
-
     assert len(set(core)) == len(core)
 
     core = set(core)
-
     assert dummy_atom not in core
 
-    # first convert to a dense graph
-    N = mol.GetNumAtoms()
+    # first convert to a dense graph, and assume that bond_idxs start from zero
+    flat_idxs = np.array(bond_idxs).flatten()
+    N = np.amax(flat_idxs) + 1
+    assert N == len(set(flat_idxs))
```

Comment:
nit: could be strengthened to 
```suggestion
    assert set(flat_idxs) == set(range(N))
```
or similar

(to rule out unexpected `-1 in flat_idxs` etc.)